frontend-react/src/components/Admin/EditSenderSettings.test.tsx

```diff
@@ -89,8 +93,52 @@ describe("EditSenderSettings", () => {
         });
 
         expect(processingTypeField).toHaveValue(testProcessingType);
-        fireEvent.click(screen.getByTestId("submit"));
+        fireEvent.click(editJsonAndSaveButton);
         fireEvent.click(screen.getByTestId("editCompareCancelButton"));
         fireEvent.click(screen.getByTestId("senderSettingDeleteButton"));
     });
+
+    describe("should validate name", () => {
+        const consoleTraceSpy = jest.fn();
+
+        beforeEach(() => {
+            jest.spyOn(console, "trace").mockImplementationOnce(
+                consoleTraceSpy
+            );
+        });
+
+        afterEach(() => {
+            jest.resetAllMocks();
+        });
+
+        test("should display an error if name value is prohibited", async () => {
+            fireEvent.change(nameField, {
+                target: { value: "Organization" },
+            });
+            expect(nameField).toHaveValue("Organization");
+
+            fireEvent.click(editJsonAndSaveButton);
+            expect(consoleTraceSpy).toHaveBeenCalled();
+        });
+
+        test("should display an error if name value contains a non-ascii char", () => {
```

Comment:
Very, very, very pedantic point, but these are technically valid ASCII characters.  Might be better to name it a "non-alphanumeric" character, but a whitespace is also non-alphanumeric 😆

Reply:
Yeah, I was wondering if "Non-url-escapable-characters" would be a better way to describe it? IDK. it's also the same syntax that's allowed for a (compiler) variable name (e.g. can't start with a number).